app/javascript/shared/components/useKeyboardShortcuts.jsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect } from "preact/hooks";
+import PropTypes from 'prop-types';
+import { h } from 'preact';
+
+/**
+ * hook that can be added to a component to listen
+ * for keyboard presses
+ *
+ * @example
+ * const shortcuts = {
+ *   "ctrl+alt+KeyG": (e) => {
+ *     e.preventDefault();
+ *     alert("Control Alt G has been pressed")
+ *   }
+ * }
+ *
+ * useKeyboardShortcuts(shortcuts, someElementOrWindowObject);
+ *
+ * @param {object} shortcuts List of keyboard shortcuts/event
+ * @param {EventTarget} [eventTarget=window] An event target.
+ *
+ */
+export function useKeyboardShortcuts(shortcuts, eventTarget = window) {
+  useEffect(() => {
+    if (!shortcuts) return;
+
+    // Return true if element is a form element
+    const isFormField = (element) => {
```

Comment:
refactor (non-blocking): `isFormField` can be a function outside of `useKeyboardShortcuts` since it does not rely on anything from `useKeyboardShortcuts`.